### PR TITLE
[asr->fortran] implement `visit_ArraySize`

### PR DESCRIPTION
**Pull request by khushi-411**
Mainly inspired by cpp implementation: https://github.com/lfortran/lfortran/blob/main/src/libasr/codegen/asr_to_cpp.cpp#L551-#L567

### CODE REVIEW

**Comment by certik** [comment]
The LLVM changes are due to using a different LLVM version. Use LLVM 11 to generate tests. We might upgrade it to some more recent version, but for now we use LLVM 11. (You can use any LLVM version to run integration tests, those we test with LLVM 11-15).

File: src/libasr/codegen/asr_to_fortran.cpp [hunk 1]
```diff
@@ -870,7 +870,20 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
...
+        std::string r = "";
+        if (x.m_dim == nullptr) {
+            // TODO: return the product of all dimensions:
+            r = "0";
+        } else {
+            if( x.m_dim ) {
+                visit_expr(*x.m_dim);
+                r += s + "-1";
+            }
+        }
```
Comment (certik) [inline src/libasr/codegen/asr_to_fortran.cpp:884]: I would write this as:
```suggestion
        std::string r = "size(" + s;
        if (x.m_dim) {
            r += ", ";
            for (int i=0; i < x.n_dim; i++) {
                visit_expr(*x.m_dim);
                r += s;
                if (i < x.n_dim-1) r += ", ";
            }
        }
        r += ")";
```

**Comment by certik** [comment]
Then add or enable a test in integration_tests that uses `size()`.

**Comment by Thirumalai-Shaktivel** [comment]
The test fails because some tests that you have enabled have errors.

Try running `lfortran integration_tests/arrays_01_size.f90 --backend=fortran` on your local machine to view the errors.
When you enable the tests, please make sure it works locally.

File: src/libasr/codegen/asr_to_fortran.cpp [hunk 2]
```diff
@@ -870,7 +870,20 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
...
+        visit_expr(*x.m_v);
+        std::string r = "size(" + s;
+        if (x.m_dim) {
+            r += ", ";
+            int dim = down_cast<ASR::Integer_t>(x.m_dim)->m_kind;
+            for(int i = 0; i < dim; i++) {
+                visit_expr(*x.m_dim);
+                r += s;
+                if (i < dim-1) r += ", ";
+            }
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/codegen/asr_to_fortran.cpp:883]: ASR::Integer_t is a `ttype` node not `expr` node (You can confirm this in ASR.asdl file).
You can analyze ASR output using --show-asr to handle this.

From ASR of arrays_01_size.f90 you can see it has `IntegerConstant`, so you have to do:
```suggestion
            visit_expr(*x.m_dim);
            r += s;
```

Also, x.m_dim can be IntegerConstant or Var. Later we have to add a check for Var as well

File: src/libasr/codegen/asr_to_fortran.cpp [hunk 3]
```diff
@@ -870,7 +870,20 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
...
+        }
+        r += ")";
+    }
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/codegen/asr_to_fortran.cpp:886]: At last, we have assign the result to s;
So, add `s = r;`

File: src/libasr/codegen/asr_to_fortran.cpp (hunk 2 above)
Comment (Thirumalai-Shaktivel) [reply src/libasr/codegen/asr_to_fortran.cpp:883]: Also, you have to visit m_kind as well.

Comment (certik) [reply src/libasr/codegen/asr_to_fortran.cpp:883]: I don't think you can do `visit_expr(*x.m_dim);`, can you? I think the original code is needed, but I can see a mistake there, it should have `visit_expr(*x.m_dim[i]);`.

Comment (Thirumalai-Shaktivel) [reply src/libasr/codegen/asr_to_fortran.cpp:883]:
```
    | ArraySize(expr v, expr? dim, ttype type, expr? value)
...
```
The dim is optional and scalar here, so I think `visit_expr(*x.m_dim)` would be enough.

**Comment by khushi-411** [comment]
Thanks, @certik and @Thirumalai-Shaktivel, for reviewing the PR. I think it's ready for your review.

**Review by certik — APPROVED** [review]
I think that this is good, thanks!

**Pull request merged**